### PR TITLE
 Add spell checker as Makefile target [v2]

### PR DESCRIPTION
**Pull request by apahim**
v2:
 - Fix some spelling errors.
 - Clean white-list.
 - `echo OK` when spell check passes.

v1: #1297 
This patch adds the `make spell` command so developers can easily
check the spelling on docstrings and comments. False positives
words should be added to spell.ignore file.

Signed-off-by: Amador Pahim <apahim@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,6 +32,7 @@ all:
 	@echo "link:       Runs 'python setup.py --develop' in all subprojects and links the needed resources"
 	@echo "clean:      Get rid of scratch, byte files and removes the links to other subprojects"
 	@echo "selfcheck:  Runs tree static check, unittests and functional tests using Avocado itself"
+	@echo "spell:      Runs spell checker on comments and docstrings (requires python-enchant)"
```
Comment (ldoktor) [inline Makefile]: Shouldn't we include this in `make check`? And if so then we should add `pyenchant` to `requirements-selftests`.

Comment (apahim) [inline Makefile]: ok, updated. Tks.

File: Makefile [hunk 2]
```diff
@@ -151,7 +152,7 @@ requirements-plugins: requirements
 smokecheck:
 	./scripts/avocado run passtest
 
-check: clean develop check_cyclical modules_boundaries
+check: clean develop check_cyclical modules_boundaries spell
```
Comment (apahim) [inline Makefile]: @ldoktor, updated with the requirements-travis.

**Comment by apahim** [comment]
Actually, let me send the v3.

**Pull request closed without merging**